static/app/utils/profiling/flamegraph/selectNearestFrame.ts

```diff
@@ -0,0 +1,147 @@
+import {FlamegraphFrame} from '../flamegraphFrame';
+
+export type DirectionX = 'left' | 'right';
+export type DirectionY = 'up' | 'down';
+export type Direction = DirectionY | DirectionX;
+/**
+ * selectNearestFrame walks a FlamegraphFrame tree the direction specified and returns the nearest
+ * FlamegraphFrame relative to the starting node.
+ *
+ * below is a gist of the walking algorithm:
+ *
+ *  up    =>
```

Comment:
It would probably be easier to follow the logic if this is inlined in the fn body where the different different branching occurs